targets/XPlatCppSdk/source/test/TestApp/PlayFabEventTest.cpp

```diff
@@ -91,32 +91,35 @@ namespace PlayFabUnit
         auto pfResponse = std::dynamic_pointer_cast<const PlayFab::PlayFabEmitEventResponse>(response);
 
-        // Handle successful event delivery.
-        if (PlayFab::PlayFabErrorCode::PlayFabErrorSuccess == pfResponse->playFabError->ErrorCode)
-        {
-            ++eventPassCount;
-            eventFailLog += pfEvent->GetName() + " was sent successfully " +
-                "in the batch #" + std::to_string(pfResponse->batchNumber) + " "
-                "of " + std::to_string(pfResponse->batch->size()) + " events. "
-                "HTTP code: " + std::to_string(pfResponse->playFabError->HttpCode) +
-                ", app error code: " + std::to_string(pfResponse->playFabError->ErrorCode) + "\n";
-
-            // Keep track of the highest batch number.
-            eventBatchMax = (pfResponse->batchNumber > eventBatchMax) ? pfResponse->batchNumber : eventBatchMax;
-        }
-        // Handle failed event delivery.
-        else
-        {
-            ++eventFailCount;
-            eventFailLog += pfEvent->GetName() + " received an error back " +
-                "in the batch #" + std::to_string(pfResponse->batchNumber) + " "
-                "of " + std::to_string(pfResponse->batch->size()) + " events. "
-                "HTTP code: " + std::to_string(pfResponse->playFabError->HttpCode) +
-                ", app error code: " + std::to_string(pfResponse->playFabError->ErrorCode) +
-                ", HTTP status: " + pfResponse->playFabError->HttpStatus +
-                ", Message: " + pfResponse->playFabError->ErrorMessage +
-                "\n";
-        }
+		if (pfResponse->playFabError != nullptr)
+		{
+			// Handle successful event delivery.
+			if (PlayFab::PlayFabErrorCode::PlayFabErrorSuccess == pfResponse->playFabError->ErrorCode)
+			{
+				++eventPassCount;
+				eventFailLog += pfEvent->GetName() + " was sent successfully " +
+					"in the batch #" + std::to_string(pfResponse->batchNumber) + " "
+					"of " + std::to_string(pfResponse->batch->size()) + " events. "
+					"HTTP code: " + std::to_string(pfResponse->playFabError->HttpCode) +
+					", app error code: " + std::to_string(pfResponse->playFabError->ErrorCode) + "\n";
+
+				// Keep track of the highest batch number.
+				eventBatchMax = (pfResponse->batchNumber > eventBatchMax) ? pfResponse->batchNumber : eventBatchMax;
+			}
+			// Handle failed event delivery.
+			else
+			{
+				++eventFailCount;
+				eventFailLog += pfEvent->GetName() + " received an error back " +
+					"in the batch #" + std::to_string(pfResponse->batchNumber) + " "
+					"of " + std::to_string(pfResponse->batch->size()) + " events. "
+					"HTTP code: " + std::to_string(pfResponse->playFabError->HttpCode) +
+					", app error code: " + std::to_string(pfResponse->playFabError->ErrorCode) +
+					", HTTP status: " + pfResponse->playFabError->HttpStatus +
+					", Message: " + pfResponse->playFabError->ErrorMessage +
+					"\n";
+			}
+		}
```

Comment:
We need to log that something failed here, because of null obj.

Reply:
You may call testContext->Fail("EmitEventCallback received an error"); when pfResponse->playFabError is null

---
In reply to: [289120073](https://github.com/PlayFab/SDKGenerator/pull/534#discussion_r289120073) [](ancestors = 289120073)